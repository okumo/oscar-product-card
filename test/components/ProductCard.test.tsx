import React from 'react'
import render, { act } from 'react-test-renderer'
import { ProductTitle } from '../../src/components'
import { ProductCard } from '../../src/components/ProductCard'
import { product1 } from '../data/products'

describe('ProductImage', () => {
    test('debe mostrar el componente correctamente', () => {
        const wrapper = render.create(
            <ProductCard product={product1}>
                {() => (
                    <ProductTitle />
                )}
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot()
    })

    test('debe incrementar el contador', () => {
        const wrapper = render.create(
            <ProductCard product={product1}>
                {({ reset, increaseBy, count }) =>
                    <>
                        <ProductTitle />
                        <button onClick={reset}>reset</button>
                        <button onClick={() => increaseBy(+1)}>+1</button>
                        <span>{count}</span>
                    </>
                }
            </ProductCard>
        )

        let tree = wrapper.toJSON();
        expect(tree).toMatchSnapshot();

        act(() => {
            (tree as any).children[2].props.onClick()
        })

        tree = wrapper.toJSON()
        expect((tree as any).children[3].children[0]).toBe("1")
    })

})