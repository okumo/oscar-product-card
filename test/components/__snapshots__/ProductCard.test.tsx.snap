// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe incrementar el contador 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Productito
  </span>
  <button
    onClick={[Function]}
  >
    reset
  </button>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <span>
    0
  </span>
</div>
`;

exports[`ProductImage debe mostrar el componente correctamente 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Productito
  </span>
</div>
`;
